### PR TITLE
Restructure the CLI and improve help.

### PR DESCRIPTION
**Pull request by jsirois**

### CODE REVIEW

**Comment by jsirois** [comment]
This builds on #13 which builds on #12. The 1st commit to review is 721adb7.

**Comment by jsirois** [comment]
I'm going to submit after submitting #13 as soon as this goes green to unwind the stack from here to there.

File: science/exe.py [hunk 1]
```diff
@@ -245,90 +267,335 @@ def parse(cls, value: str) -> AppInfo:
...
+        """\
+        Map paths to files defined in your manifest.
+
+        Science looks fore each non-lazy file you define at the path denoted by its name relative
+        to the CWD you invoke science from. If any file is not at that path, you can tell science
+        to look elsewhere with: `--file <name>=<location>`.
+
+        For example, for this manifest snippet:
+
+        \b
```
Comment (jsirois) [inline science/exe.py:301]: N.B.: This is a documented click trick to force the following paragraph to be rendered without re-wrapping. Used here and elsewhere for code blocks.

File: science/exe.py [hunk 2]
```diff
@@ -245,90 +267,335 @@ def parse(cls, value: str) -> AppInfo:
...
+@dataclass(frozen=True)
+class LiftConfig:
+    file_mappings: tuple[FileMapping, ...] = ()
+    invert_lazy_ids: frozenset[str] = frozenset()
+    include_provenance: bool = False
+    app_info: tuple[AppInfo, ...] = ()
+    app_name: str | None = None
+
+
+pass_lift = click.make_pass_decorator(LiftConfig)
```
Comment (jsirois) [inline science/exe.py:279]: This `click.make_pass_decorator` paired with `click.pass_context` below on the _lift group is a documented means to support structured chaining of sub-command configuration from the parent group(s) with sane isolation of parent group flags from sub-command knowledge.

**Review by benjyw — COMMENTED** [review]
LGTM

**Pull request merged**